code/modules/spells/spell_types/touch/_touch.dm

```diff
@@ -170,24 +170,59 @@
 	SHOULD_NOT_OVERRIDE(TRUE) // DEFINITELY don't put effects here, put them in cast_on_hand_hit
 
 	if(!can_hit_with_hand(target, caster))
-		return
+		return NONE
+
+	SEND_SIGNAL(src, COMSIG_SPELL_TOUCH_HAND_HIT, target, caster, source)
+
+	var/mob/mob_victim = target
+	if(istype(mob_victim) && mob_victim.can_block_magic(antimagic_flags))
+		on_antimagic_triggered(source, target, caster)
 
-	INVOKE_ASYNC(src, PROC_REF(do_hand_hit), source, target, caster)
```

Comment:
If it doesn't actually need to be asyncronous, then just keep the proc `do_hand_hit` and call it normally

Reply:
Can't do that because do_hand_hit() sleeps

Reply:
Evidently it doesn't all sleep since you can move all the code contained in `do_hand_hit` to this proc which forbids sleeping